contract/src/test/EthRadicleCloudTest.t.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.6;
+
+import "ds-test/test.sol";
+import {Hevm} from "./Hevm.t.sol";
+import "../EthRadicleCloud.sol";
+
+contract EthRadicleCloudTest is DSTest {
+    EthRadicleCloud private rc;
+
+    function setUp() public {
+        // price    = 1 wei/block
+        // duration = 200 blocks
+        // owner    = address(this)
+        rc = new EthRadicleCloud(1 wei, 200, address(this));
+    }
+
+    // contract is owner thus should be able to receive ether
+    receive() external payable {
+        msg.value;
+    }
+
+    function buyOnePackageForAddress(uint160 org) public {
+        rc.buyOrRenew{value: rc.getPrice()}(address(org), address(org));
+    }
+
+    function testPrice() public {
+        uint256 price = rc.getPrice();
+        assertEq(price, 1 wei * rc.duration());
+    }
+
+    function testFailNewPurchaseLowAmount() public {
+        rc.buyOrRenew{value: rc.getPrice() - 1}(address(2), address(2));
+        assertEq(rc.getExpiry(address(2)), rc.duration());
+    }
+
+    function testNewPurchase() public {
+        buyOnePackageForAddress(2);
+        assertEq(rc.getExpiry(address(2)), rc.duration());
+    }
+
+    function testRenew() public {
+        buyOnePackageForAddress(2);
+        buyOnePackageForAddress(2);
+        assertEq(rc.getExpiry(address(2)), rc.duration() * 2);
+    }
+
+    function testSinglePurchaseMid() public {
+        Hevm(HEVM_ADDRESS).roll(20);
+        buyOnePackageForAddress(2);
+        Hevm(HEVM_ADDRESS).roll(120);
+        assertEq(rc.withdrawRealizedRevenue(), 100);
+    }
+
+    function testSinglePurchaseEnd() public {
+        Hevm(HEVM_ADDRESS).roll(20);
+        buyOnePackageForAddress(2);
+        Hevm(HEVM_ADDRESS).roll(220);
+        assertEq(rc.withdrawRealizedRevenue(), 200);
+    }
+
+    function testSinglePurchaseOut() public {
+        Hevm(HEVM_ADDRESS).roll(20);
+        buyOnePackageForAddress(2);
+        Hevm(HEVM_ADDRESS).roll(999);
+        assertEq(rc.withdrawRealizedRevenue(), 200);
+    }
+
+    function testSinglePurchaseMultiWithdraw() public {
+        Hevm(HEVM_ADDRESS).roll(20);
+        buyOnePackageForAddress(2);
+        Hevm(HEVM_ADDRESS).roll(120);
+        assertEq(rc.withdrawRealizedRevenue(), 100);
+        Hevm(HEVM_ADDRESS).roll(220);
+        assertEq(rc.withdrawRealizedRevenue(), 100);
+        Hevm(HEVM_ADDRESS).roll(300);
```

Comment:
It's cleaner to have `Hevm hevm = Hevm(HEVM_ADDRESS);` as a contract variable and then just call `hevm.roll(...)`

Reply:
https://github.com/radicle-dev/radicle-cloud/pull/1/commits/5e4baff52f37627e011a62ae72166b0930007670